### PR TITLE
Med 42 implement login api

### PR DESCRIPTION
**Pull request by SoulOfDarknes**
add Login (email, password):

 1. Login doctor  
 2. Make validation if email and password match
 3.  Add guard and passport with jwt modules

PR checklist:

1.1 Common
- [x] 1.1.1 types for input and output params
- [x] 1.1.2 no any, should be also added to eslint rules
- [x] 1.1.3 try/catch for any async function
- [x] 1.1.4 no magic numbers
- [x] 1.1.5 compare only with constants not with strings (instead of user === ‘admin’, better user === roles.admin)
- [x] 1.1.6 no ternary operator inside ternary operator (bad example: question ? first: second ? third: fourth)
- [x] 1.1.7 avoid similar types with duplicating by generics
- [x] 1.1.8 no console.log in pr
- [x] 1.1.9 .env file should be in .gitignore
- [x] 1.1.10 delete commented code if it's not part of documentation
- [x] 1.1.11 no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file
- [x] 1.1.12 constants and function names should be lowercase.
- [x] 1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant variable
- [x] 1.1.14 import rules. imports should come in a specific order: node modules, absolute path, relative path
- [x] 1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’, const EACH_DAY_15h_15min = ‘15 3 * * *’

1.2 Backend
- [x] 1.2.1 swagger for each endpoint
- [x] 1.2.2 less requests to db, all data should be taken with one query
- [x] 1.2.3 public in method use only is use function externally
- [x] 1.2.4 use ConfigService instead of process.env

1. Get a Token 
image (3)
image

2. Wrong password
image (4)

3. Wrong email
image (2)

4. Using Guards with token 
image (5)

5. If token is wrong on Guard
image (1)

### CODE REVIEW

File: src/filters/gobalException.filter.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export class GlobalExceptionFilter implements ExceptionFilter {
 
     let message: string | string[];
     let status: HttpStatus;
-
+    console.log(exception);
     switch (true) {
```
Comment (lyudad) [inline src/filters/gobalException.filter.ts]: its strange condition, I dont think its right

File: src/filters/gobalException.filter.ts [hunk 2]
```diff
@@ -21,7 +21,7 @@ export class GlobalExceptionFilter implements ExceptionFilter {
 
     let message: string | string[];
     let status: HttpStatus;
-
+    console.log(exception);
```
[the remark this answers is not in the document]
Comment (SoulOfDarknes) [reply src/filters/gobalException.filter.ts:24]: done

[the remark this answers is not in the document]
Comment (SoulOfDarknes) [reply src/filters/gobalException.filter.ts]: done

File: src/filters/gobalException.filter.ts (hunk 1 above)
Comment (SoulOfDarknes) [reply src/filters/gobalException.filter.ts]: this filter was agreed with you a little earlier

File: src/modules/auth/auth.controller.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  @ApiResponse({
+    status: 200,
+    description: 'Doctor was logged in',
+  })
+  @ApiResponse({
+    status: 401,
+    description: 'Invalid email or password',
+  })
+
+  async login(@Body() dto: LoginDoctorDto) {
```
[the remark this answers is not in the document]
Comment (SoulOfDarknes) [reply src/modules/auth/auth.controller.ts]: I don't see what the problem is, but I assume it's "try - catch" but we have a filter for that

File: src/filters/gobalException.filter.ts (hunk 1 above)
Comment (lyudad) [reply src/filters/gobalException.filter.ts]: then I misunderstand something, I could not approve `switch (true) {`

File: src/modules/auth/auth.controller.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (lyudad) [reply src/modules/auth/auth.controller.ts]: its not "try - catch", you have not follow checklist, you don't have types

File: src/filters/gobalException.filter.ts (hunk 1 above)
Comment (SoulOfDarknes) [reply src/filters/gobalException.filter.ts]: done

File: src/modules/auth/auth.controller.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (SoulOfDarknes) [reply src/modules/auth/auth.controller.ts]: done

File: src/modules/auth/auth.service.ts [hunk 4]
```diff
@@ -20,6 +21,43 @@ export class AuthService {
     private config: ConfigService,
   ) {}
 
+  async login(dto: LoginDoctorDto) {
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: types missed. go through the checklist

Comment (SoulOfDarknes) [reply src/modules/auth/auth.service.ts]: done

**Review by lyudad — APPROVED** [review]

**Pull request merged**